internal/common/baseclient/baseclient_iigo.go

```diff
@@ -0,0 +1,42 @@
+package baseclient
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+)
+
+// CommonPoolResourceRequest is called by the President in IIGO to
+// request an allocation of resources from the common pool.
+func (c *BaseClient) CommonPoolResourceRequest() int {
+	return 20
+}
+
+// ResourceReport is an island's self-report of its own resources.
+func (c *BaseClient) ResourceReport() int {
+	return c.clientGameState.ClientInfo.Resources
+}
+
+// RuleProposal is called by the President in IIGO to propose a
+// rule to be voted on.
+func (c *BaseClient) RuleProposal() string {
+	allRules := rules.AvailableRules
+	for k, _ := range allRules {
+		return k
+	}
+	return ""
+}
+
+// GetClientPresidentPointer is called by IIGO to get the client's implementation of the President Role
+func (c *BaseClient) GetClientPresidentPointer() roles.President {
```

Comment:
Clients will be allowed to implement some functions (All that are exposed in the Role interfaces) so that they can put some strategy in and inf;uence how IIGO runs, to keep the data-types and stuff standardised we require them to implement the appropriate interface and pass us their custom version which we run the functions from. Note: All exposed functions are Decision functions, Server still actions them so the clients can't mess with anything they aren't meant to. They can only cheat.

Reply:
So this is why all of the base_speaker, judge and president implementation are in the server? Could they simply be called speaker, judge and president then, to avoid naming confusion. Considering that the baseClient could/should in theory be completely overwritten but the server side implementation you have written should not?